include/mfmg/dealii/amge_host.templates.hpp

```diff
@@ -128,7 +128,8 @@ AMGe_host<dim, MeshEvaluator, VectorType>::compute_local_eigenvectors(
   std::vector<dealii::Vector<double>> eigenvectors(
       n_eigenvectors, dealii::Vector<double>(n_dofs_agglomerate));
 
-  if (_eigensolver_type == "arpack")
+  auto eigensolver_type = _eigensolver_params.get("type", "arpack");
```

Comment:
`const` it.  Also I would feel more comfortable if you specified `<std::string>`.  I am not sure what the deduction rule is.  I sure hope you don't get a `char *`...

Reply:
(By that I mean `ptree::get<std::string>(...)`)

Reply:
Quick try in Cling says the return type is `boost::enable_if<boost::property_tree::detail::is_character<char>, std::basic_string<char, std::char_traits<char>, std::allocator<char> > >::type` so you're good :/